src/cors.c

```diff
@@ -80,25 +80,27 @@ static bool is_origin_allowed(cors_config *c, char *origin) {
  * user-defined allow list
  */
 static bool is_method_allowed(cors_config *c, char *method) {
-  if (!has_elements(c->allowed_methods)) {
```

Comment:
Technically, the original implementation avoids unnecessary computation by returning early as well, but I'm amenable to nesting them since it gets us less return statements.